### ISSUE TITLE
Add sleep timer to sleep desktop/device after X idle time

### ISSUE THREAD

**Issue by ziggimon** [plexinc/plex-media-player#22]
With firstly the embedded builds in mind we should add a sleep timer in settings so a user can define X amount of idle time before PMP sleeps the device/computer its running on. Default should be off.

We have the sleep option on Embedded. We simply need to add a timer that then invokes the sleep action.

**Comment by dogonwheels** [comment]
Always love your 'simply's :laughing: 

Seems a sensible addition.

**Comment by tru** [comment]
Agreed this should be easy to implement as well. I wonder if this is something we want to do in the web-client or in the host. We can hook it up in both places. Opinions @mseeley @firasdib?

**Comment by luigira** [comment]
PLEASE!!!!

**Comment by markus101** [comment]
It seems reasonable. I think it should live on the host side, though web would have to do some work to add the settings.

**Comment by LongChair** [comment]
Well we could reimplement this on host side, but we already have the mechanism for screensaver right ? So would be just to have that setting and call `api.system.suspend()` after another <settings> seconds right ?

**Comment by markus101** [comment]
I believe the screensaver is all done host side, but the web app tells the host to enable or disable it.

**Comment by LongChair** [comment]
I’m pretty sure the screensaver is in web client ;)

**Comment by markus101** [comment]
I'm looking at a call we're making, `konvergoChannel.api.power.setScreensaverEnabled(enable);` (where enable is true or false), is that for the OS screensaver?

**Comment by LongChair** [comment]
Yeah this is for System screensaver and on some platforms it’s not doing anything (OS X).

I’m referring to the app screensaver (Plex logo bouncing around)

**Comment by markus101** [comment]
Can we defer to the OS to handle this? In the case of embedded we could expose a setting for it, but Eric and I agree that having the app control it doesn't seem like the right thing to do. The Web app will tell the host to disable the OS screen saver, can that be linked to tell it to disable sleep as well?

**Comment by markus101** [comment]
Focusing only on embedded it may make sense for this to be handled in the web app, similar to the screensaver, but not something we can prioritize working on currently.

On the host side, is there (or can one be added) that the web app can call to sleep the system?
We'll also need a setting to sleep the system, unfortunately we chose `sleepTime` for the screensaver in the past, maybe `systemSleepTime` with 0 to disable?

**Comment by LongChair** [comment]
I think we already have such method, system.sleep() maybe ?

For the setting name yeah that is fine.
Totally understand that this is not priority at the moment :)
